From 75a842c7a91084d15544b2e42a201a9c859cbe4d Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 30 Dec 2022 00:42:46 +0530
Subject: [PATCH 23/25] TODO: gutils: Hard-code windows version detection in
 g_get_os_info

This should use GetVersionExA instead, which is available on UWP.
---
 glib/gutils.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index 3beaf8e..dd23821 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -1182,6 +1182,8 @@ g_set_application_name (const gchar *application_name)
 }
 
 #ifdef G_OS_WIN32
+
+#ifndef G_WINAPI_ONLY_APP
 /* For the past versions we can just
  * hardcode all the names.
  */
@@ -1298,6 +1300,7 @@ get_windows_8_1_update (void)
 
   return result;
 }
+#endif /* !G_WINAPI_ONLY_APP */
 
 static gchar *
 get_windows_version (gboolean with_windows)
@@ -1305,6 +1308,10 @@ get_windows_version (gboolean with_windows)
   GString *version = g_string_new (NULL);
   gboolean is_win_server = FALSE;
 
+#ifdef G_WINAPI_ONLY_APP
+  /* FIXME: Use GetVersionExA instead of hard-coding this */
+  g_string_append (version, "10");
+#else
   if (g_win32_check_windows_version (10, 0, 0, G_WIN32_OS_ANY))
     {
       gchar *win10_release;
@@ -1443,13 +1450,14 @@ get_windows_version (gboolean with_windows)
 
       return NULL;
     }
+#endif /* G_WINAPI_ONLY_APP */
 
   if (with_windows)
     g_string_prepend (version, "Windows ");
 
   return g_string_free (version, FALSE);
 }
-#endif
+#endif /* G_OS_WIN32 */
 
 #if defined (G_OS_UNIX) && !defined (__APPLE__)
 static gchar *
-- 
2.38.1

